pkg/disk/partition_table.go

```diff
@@ -894,3 +894,145 @@ func (pt *PartitionTable) GetMountpointSize(mountpoint string) (uint64, error) {
 
 	panic(fmt.Sprintf("no sizeable of the entity path for mountpoint %s, this is a programming error", mountpoint))
 }
+
+// EnsureRootFilesystem adds a root filesystem if the partition table doesn't
+// already have one.
+//
+// When adding the root filesystem, add it to:
+//
+//   - The first LVM Volume Group if one exists, otherwise
+//   - The first Btrfs volume if one exists, otherwise
+//   - At the end of the plain partitions.
+//
+// For LVM and Plain, the fsType argument must be a valid filesystem type.
+func EnsureRootFilesystem(pt *PartitionTable, fsType FSType) error {
```

Comment:
(super nitpick) maybe defaultFsType because this will not "ensure" that the fsType is of the given type just what to use if it's missing